docs/tools/plist.md

```diff
@@ -0,0 +1,301 @@
+			<string>report-converter-type</string>
+		</dict>
+	</array>
+	<key>files</key>
+	<array>
+		<string>files/file.cpp</string>
+	</array>
+	<key>metadata</key>
+	<dict>
+		<key>analyzer</key>
+		<dict>
+			<key>name</key>
+			<string>...</string>
+		</dict>
+		<key>generated_by</key>
+		<dict>
+			<key>name</key>
+			<string>report-converter</string>
+			<key>version</key>
+			<string>x.y.z</string>
+		</dict>
+	</dict>
+</dict>
+</plist>
+```
+
+### Structure of an Empty .plist File
+
+```xml
+<plist version="1.0">
+<dict>
+	<key>diagnostics</key>
+	<array>
+	</array>
+	<key>files</key>
+	<array>
+	</array>
+</dict>
+</plist>
+```
+
+---
+
+## Diagnostic Entry Examples
+
+The `.plist` file may contain the following kinds of entries:
+
+### 1. `bug_path_event` — Bug Path Events
+
+Bug Path Events are the core of how CodeChecker explains the progression of an 
+issue in the code. They show what happened at each step that led to the 
+reported bug, in a step-by-step trace.
+
+```xml
+<key>path</key>
+<array>
+  <dict>
+    <key>edges</key>
+    <array>
+      <dict>
+        <key>end</key>
+        <array>
+          <dict>
+            <key>col</key>
+            <integer>1</integer>
+            <key>file</key>
+            <integer>1</integer>
+            <key>line</key>
+            <integer>3</integer>
+          </dict>
+          <dict>
+            <key>col</key>
+            <integer>1</integer>
+            <key>file</key>
+            <integer>1</integer>
+            <key>line</key>
+            <integer>3</integer>
+          </dict>
+        </array>
+        <key>start</key>
+        <array>
+          <dict>
+            <key>col</key>
+            <integer>1</integer>
+            <key>file</key>
+            <integer>0</integer>
+            <key>line</key>
+            <integer>19</integer>
+          </dict>
+          <dict>control
+            <key>col</key>
+            <integer>1</integer>
+            <key>file</key>
+            <integer>0</integer>
+            <key>line</key>
+            <integer>19</integer>
+          </dict>
+        </array>
+      </dict>
+    </array>
+    <key>kind</key>
+    <string>control</string>
+  </dict>
+  <dict>
+    <key>depth</key>
+    <integer>0</integer>
+    <key>kind</key>
+    <string>event</string>
+    <key>location</key>
+    <dict>
+      <key>col</key>
+      <integer>1</integer>
+      <key>file</key>
+      <integer>0</integer>
+      <key>line</key>
+      <integer>19</integer>
+    </dict>
+    <key>message</key>
+    <string>First bug path event</string>
+  </dict>
+  <dict>
+    <key>depth</key>
+    <integer>0</integer>
+    <key>kind</key>
+    <string>event</string>
+    <key>location</key>
+    <dict>
+      <key>col</key>
+      <integer>1</integer>
+      <key>file</key>
+      <integer>1</integer>
+      <key>line</key>
+      <integer>3</integer>
+    </dict>
+    <key>message</key>
+    <string>Second bug path event</string>
+  </dict>
+</array>
+```
+
```

Comment:
Please describe the meaning of each keys.

what are  the kind types?
what is the meaning of depth? (how it is visualized?)